osu.Game/Online/Multiplayer/StatefulMultiplayerClient.cs

```diff
@@ -534,5 +568,25 @@ private void updateUserPlayingState(int userId, MultiplayerUserState state)
             else
                 CurrentMatchPlayingUserIds.Remove(userId);
         }
+
+        private Task scheduleAsync(Action action)
```

Comment:
I didn't implement this as an extension method because this is quite unsafe due to `async () =>` which probably requires something like https://github.com/ppy/osu-framework/pull/4149 to be handled properly.

Reply:
can you explain where the `async () =>` is? having trouble understanding this comment.

Reply:
I'm thinking this would be a pretty handy method to have in `Scheduler` itself (or just make our existing scheduler delegate stuff into `Task`s finally)

Reply:
There is no `async () =>` here, because all usages have made sure not to do so. My comment was intended to read that _it's possible_ for a user to write such a lambda if this were moved to an extension method, at which point it needs proper handling.

Reply:
I see. Would you be against this being added to `Scheduler` itself in some way?

Reply:
No I'm not, so long as it's handled properly. I agree it's a useful thing and I originally had it as an extension method before realising its shortcomings.